src/app/plugins/stock.plugin.ts

```diff
@@ -0,0 +1,99 @@
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType, IHttpResponse } from '../../common/types';
+import Environment from '../../environment';
+
+export class StockPlugin extends Plugin {
+  public name: string = 'Stock Plugin';
+  public description: string = 'Get stock quotes';
+  public usage: string = 'stock <ticker>; stock AAPL';
+  public permission: ChannelType = ChannelType.Public;
+
+  private _API_URL: string = 'https://www.alphavantage.co/query?';
+  private _ADVANCED_QUOTE_LINK: string = 'https://finance.yahoo.com/quote/';
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public validate(message: IMessage, args: string[]) {
+    // in order to error handle
+    return args.length > 0;
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    if (!args || args.length < 1) {
+      message.reply('Bad query.');
+      return;
+    }
+
+    const ticker = args[0];
+
+    const call_url =
+      this._API_URL +
+      'function=GLOBAL_QUOTE' +
+      '&symbol=' +
+      ticker +
+      '&apikey=' +
+      Environment.StockApiToken;
+
+    const data = await this.container.httpService
+      .get(call_url)
+      .then((res: IHttpResponse) => {
+        return res.data;
+      })
+      .catch((err) => {
+        console.log(err);
+        return new Object();
```

Comment:
why are we returning a new object here?

Reply:
the `then` returns an object, so i want to be consistent